components/dashboard/src/settings/ProfileInformation.tsx

```diff
@@ -0,0 +1,202 @@
+        return !(
+            !!user.additionalData.lastUpdatedDetailsNudge &&
+            isDateSmaller(hoursBefore(new Date().toISOString(), 24 * 365), user.additionalData.lastUpdatedDetailsNudge)
+        );
+    }
+
+    /**
+     * @param state
+     * @returns error message or empty string when valid
+     */
+    export function validate(state: ProfileState): string {
+        if (state.name.trim() === "") {
+            return "Name must not be empty.";
+        }
+        if (state.email.trim() === "") {
+            return "Email must not be empty.";
+        }
+        if (!/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(state.email.trim())) {
+            return "Please enter a valid email.";
+        }
+        return "";
+    }
+
+    export function NudgeForProfileUpdateModal() {
+        const { user, setUser } = useContext(UserContext);
+        const original = ProfileState.getProfileState(user!);
+        const [profileState, setProfileState] = useState(original);
+        const [errorMessage, setErrorMessage] = useState("");
+        const [visible, setVisible] = useState(shouldNudgeForUpdate(user!));
+
+        const saveProfileState = () => {
+            const error = ProfileState.validate(profileState);
+            setErrorMessage(error);
+            if (error) {
+                return;
+            }
+            const updatedUser = ProfileState.setProfileState(user!, profileState);
+            setUser(updatedUser);
+            getGitpodService().server.updateLoggedInUser(updatedUser);
+            setVisible(shouldNudgeForUpdate(updatedUser!));
+        };
+
+        const cancelProfileUpdate = () => {
+            setProfileState(original);
+            saveProfileState();
+        };
+
+        return (
+            <Modal
+                title="Update Your Profile Information"
+                visible={visible}
+                onClose={cancelProfileUpdate}
+                closeable={true}
+                className="_max-w-xl"
+                buttons={
+                    <div>
+                        <button className="secondary" onClick={cancelProfileUpdate}>
+                            Cancel
+                        </button>
+                        <button onClick={saveProfileState}>Save</button>
+                    </div>
+                }
+            >
+                <ProfileInformation
+                    profileState={profileState}
+                    errorMessage={errorMessage}
+                    updated={false}
+                    setProfileState={setProfileState}
+                />
+            </Modal>
+        );
+    }
+}
+
+export default function ProfileInformation(props: {
+    profileState: ProfileState.ProfileState;
+    setProfileState: (newState: ProfileState.ProfileState) => void;
+    errorMessage: string;
+    updated: boolean;
+    children?: React.ReactChild[] | React.ReactChild;
+}) {
+    return (
+        <div>
+            <h3>Profile</h3>
```

Comment:
**suggestion**: In case we leave this section here as is, regardless if we simplify it, using the same heading style here is hurting information hierarchy as it conflicts with the modal title. Ideally, dropping this in the modal would be great.